### PR TITLE
[wip] mass debug

### PR DESCRIPTION
**Pull request by oksanaguba**
for browsing

### CODE REVIEW

File: components/eamxx/src/control/atmosphere_driver.cpp [hunk 1]
```diff
@@ -924,17 +931,18 @@ void AtmosphereDriver::set_initial_conditions ()
...
-    m_atm_logger->debug("    [EAMxx] Reading fields from file ...");
+    m_atm_logger->info("    [EAMxx] Reading fields from file ...");
     const auto& file_name = ic_pl.get<std::string>("Filename");
     for (const auto& it : m_field_mgrs) {
       const auto& grid_name = it.first;
+      //m_atm_logger->info("grd name for FM is "+grid_name);
       read_fields_from_file (ic_fields_names[grid_name],it.first,file_name,m_current_ts);
     }
-    m_atm_logger->debug("    [EAMxx] Reading fields from file ... done!");
+    m_atm_logger->info("    [EAMxx] Reading fields from file ... done!");
```
Comment (AaronDonahue) [inline components/eamxx/src/control/atmosphere_driver.cpp:941]: You can have all "debug" messages print by changing the atm_log_level to "debug"
```
./atmchange atm_log_level=debug
```

**Comment by oksanaguba** [comment]
For the latest commit with changes to make shoc and p3 operate with wet mmrs and wet pdel (see //OG CHANGE in shor and p3 interface files), for ne4np4 grid and 100 physics time steps, physics loop (mac_aero_mic,rrtmgp) produces
```
[ac.onguba@chrlogin1 cmake-bld]$ zgrep "OG phys" ../../run/e3sm.log.283057.230209-222827.gz 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000481 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.000000000000721 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000279451360542 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000085 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000130 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.000000000000137 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.000000000000040 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000140 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000154 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000150 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000215 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000227 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000067 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000187 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.000000000000162 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000278 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000000000000307
...
```
with the only outlier in the 3rd entry. From more fine-grained output, the outlier is shoc, so, the fixes listed below are not enough, there is something else going on, too. Without the fixes, the output looks like this:
```
[ac.onguba@chrlogin1 cmake-bld]$ zgrep "OG phys" ../../run/e3sm.log.283066.230209-232922 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], -0.000233378477130 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.000993730197650 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.000710078750727 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.001062044545426 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.001114400414846 
0: OG phys [qflx*dt - precip] - [wsum_after - wsum_before], 0.001100666784154 
...
```

The fixes in shoc and p3 are most likely due to:
-- shoc with dry mmr not converting qflx wrt dry density
-- shoc with dry mmr not using pseudo_density_dry for tracer mixing
-- for p3, we haven't arrived at the conclusion why the fixes worked, since p3 does use dry pseudodensity. one suggestion was to turn off sedimentation.

I will do a run without sedimentation for p3, otherwise, maybe other people will have ideas how to proceed. There is more to investigate -- an outlier sample in shoc, possible issues with computation of dz in shoc if dry mmrs are to be used (more details are on slack).

**Comment by oksanaguba** [comment]
Running the original P3 without cloud/ice/rain sedimentation does not seem to conserve mass. I see that precip is now zero, but conservation errors are back `../../run/e3sm.log.283330.230210-125314` .

**Comment by oksanaguba** [comment]
closing to remove it from the list, this PR is needed for future conservation work

**Pull request closed without merging**